python/mxnet/profiler.py

```diff
@@ -35,13 +68,16 @@ def profiler_set_config(mode='symbolic', filename='profile.json'):
     filename : string, optional
         The name of output trace file. Defaults to 'profile.json'.
     """
-    mode2int = {'symbolic': 0, 'all': 1}
-    check_call(_LIB.MXSetProfilerConfig(
-        ctypes.c_int(mode2int[mode]),
-        c_str(filename)))
+    warnings.warn('profiler.profiler_set_config() is deprecated. ' \
+                  'Please use profiler.set_config() instead')
+    keys = c_str_array([key for key in ["profile_" + mode, "filename"]])
+    values = c_str_array([str(val) for val in [True, filename]])
+    assert len(keys) == len(values)
+    check_call(_LIB.MXSetProfilerConfig(len(keys), keys, values))
+
 
-def profiler_set_state(state='stop'):
```

Comment:
This is a breaking change. The previous API should not be renamed. At least an alias should be kept.

Reply:
That change was suggested by Eric, so please take that up with him

Reply:
@piiswrong we should keep backward compatibility

Reply:
Agreed with @eric-haibin-lin. Let's do the right thing.

Reply:
PR re-adding profiler_set_state(): https://github.com/apache/incubator-mxnet/pull/10156